robot/framesystem/framesystem.go

```diff
@@ -131,25 +143,38 @@ func (svc *frameSystemService) Reconfigure(ctx context.Context, deps resource.De
 	components := make(map[string]resource.Resource)
 	for name, r := range deps {
 		short := name.ShortName()
-		// is this only for InputEnabled components or everything?
 		if _, present := components[short]; present {
-			return DuplicateResourceShortNameError(short)
+			svc.parts = nil
+			svc.lastReconfigureError = DuplicateResourceShortNameError(short)
+			return svc.lastReconfigureError
 		}
 		components[short] = r
 	}
 	svc.components = components
 
 	fsCfg, err := resource.NativeConfig[*Config](conf)
 	if err != nil {
+		svc.parts = nil
+		svc.lastReconfigureError = err
+		return err
+	}
+
+	parts, err := svc.ConfigToParts(fsCfg)
+	if err != nil {
+		svc.parts = nil
+		svc.lastReconfigureError = err
 		return err
 	}
 
-	sortedParts, err := referenceframe.TopologicallySortParts(fsCfg.Parts)
+	sortedParts, err := referenceframe.TopologicallySortParts(parts)
```

Comment:
This will return a `NewParentFrameMissingError` in the case on line 127 above, where one component is not yet ready but perhaps one of its children is.

This should be eventually consistent, but I believe that as written it is now impossible to get even a partial framesystem if any part with children is offline.

Reply:
Yeah, partial framesystems are never something that we've supported to my knowledge, and this does not attempt to fix this.